src/coreclr/System.Private.CoreLib/src/System/Reflection/MdImport.cs

```diff
@@ -470,86 +571,102 @@ public ConstArray GetMemberRefProps(int memberTokenRef)
             out int constructorToken,
             out ConstArray signature)
         {
-            _GetCustomAttributeProps(m_metadataImport2, customAttributeToken,
-                out constructorToken, out signature);
+            int hr = GetCustomAttributeProps(m_metadataImport2, customAttributeToken, out constructorToken, out signature);
+            if (hr < 0)
+            {
+                throw new BadImageFormatException();
+            }
         }
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern void _GetClassLayout(IntPtr scope,
-            int typeTokenDef, out int packSize, out int classSize);
+        private static extern int GetClassLayout(IntPtr scope, int typeTokenDef, out int packSize, out int classSize);
+
         public void GetClassLayout(
             int typeTokenDef,
             out int packSize,
             out int classSize)
         {
-            _GetClassLayout(m_metadataImport2, typeTokenDef, out packSize, out classSize);
+            int hr = GetClassLayout(m_metadataImport2, typeTokenDef, out packSize, out classSize);
+            if (hr < 0)
+            {
+                throw new BadImageFormatException();
+            }
         }
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern bool _GetFieldOffset(IntPtr scope,
-            int typeTokenDef, int fieldTokenDef, out int offset);
+        private static extern bool GetFieldOffset(IntPtr scope, int typeTokenDef, int fieldTokenDef, out int offset, out int hr);
```

Comment:
It looks odd to return the HRESULT in out arg. Switch the order so that the method returns HRESULT?

Reply:
Done.